posthog-core/src/index.ts

```diff
@@ -707,12 +707,15 @@ export abstract class PostHogCore extends PostHogCoreStateless {
     return this.getPersistedProperty<string>(PostHogPersistedProperty.DistinctId) || this.getAnonymousId()
   }
 
-  register(properties: { [key: string]: any }): void {
+  register(properties: { [key: string]: any }, options?: { persist: boolean }): void {
```

Comment:
Maybe I'm missing something but if I do:

1. `register({non_persisted_property: 'xyz'}, {persist: false})`
2. `register({persisted_property: 'xyz'})`

Then it will persist the non-persisted property too?

Reply:
Ooooohhh good catch! Yep. Looks like we need a separate non-persisted props object...